src/test/unit/mcmc/hmc/hamiltonians/dense_e_metric_test.cpp

```diff
@@ -29,114 +29,110 @@ TEST(McmcDenseEMetric, sample_p) {
 
   for (int i = 0; i < n_samples; ++i) {
     metric.sample_p(z, base_rng);
-    double T = metric.T(z);
+    double tau = metric.tau(z);
 
-    double delta = T - m;
+    double delta = tau - m;
     m += delta / static_cast<double>(i + 1);
-    m2 += delta * (T - m);
+    m2 += delta * (tau - m);
   }
-  
+
   double var = m2 / (n_samples + 1.0);
-  
+
   // Mean within 5sigma of expected value (d / 2)
-  EXPECT_EQ(true, fabs(m   - 0.5 * q.size()) < 5.0 * sqrt(var));
-  
+  EXPECT_TRUE(std::fabs(m   - 0.5 * q.size()) < 5.0 * sqrt(var));
+
   // Variance within 10% of expected value (d / 2)
-  EXPECT_EQ(true, fabs(var - 0.5 * q.size()) < 0.1 * q.size());
+  EXPECT_TRUE(std::fabs(var - 0.5 * q.size()) < 0.1 * q.size());
 }
 
-TEST(McmcDenseEMetric, gradients) {  
+TEST(McmcDenseEMetric, gradients) {
   rng_t base_rng(0);
-  
+
   Eigen::VectorXd q = Eigen::VectorXd::Ones(11);
-  
+
   stan::mcmc::dense_e_point z(q.size());
   z.q = q;
   z.p.setOnes();
-  
+
   std::fstream data_stream(std::string("").c_str(), std::fstream::in);
   stan::io::dump data_var_context(data_stream);
   data_stream.close();
-  
+
 
   std::stringstream model_output, metric_output;
   stan::interface_callbacks::writer::stream_writer writer(metric_output);
   std::stringstream error_stream;
   stan::interface_callbacks::writer::stream_writer error_writer(error_stream);
 
-
   funnel_model_namespace::funnel_model model(data_var_context, &model_output);
 
   stan::mcmc::dense_e_metric<funnel_model_namespace::funnel_model, rng_t> metric(model);
-  
+
   double epsilon = 1e-6;
-  
-  metric.update(z, writer, error_writer);
-  
-  Eigen::VectorXd g1 = metric.dtau_dq(z);
-  
+
+  metric.init(z, writer, error_writer);
+  Eigen::VectorXd g1 = metric.dtau_dq(z, writer, error_writer);
+
   for (int i = 0; i < z.q.size(); ++i) {
-    
+
     double delta = 0;
-    
+
     z.q(i) += epsilon;
-    metric.update(z, writer, error_writer);
+    metric.update_potential(z, writer, error_writer);
     delta += metric.tau(z);
-    
+
     z.q(i) -= 2 * epsilon;
-    metric.update(z, writer, error_writer);
+    metric.update_potential(z, writer, error_writer);
     delta -= metric.tau(z);
-    
+
     z.q(i) += epsilon;
-    metric.update(z, writer, error_writer);
-    
+    metric.update_potential(z, writer, error_writer);
+
     delta /= 2 * epsilon;
-    
+
     EXPECT_NEAR(delta, g1(i), epsilon);
-    
   }
-  
+
   Eigen::VectorXd g2 = metric.dtau_dp(z);
-  
+
   for (int i = 0; i < z.q.size(); ++i) {
-    
+
```

Comment:
(just a test, so no need to fix)  No newlines at beginnings of blocks --- cpplint will flag it.